nesta/packages/crunchbase/crunchbase_collect.py

```diff
@@ -0,0 +1,333 @@
+from nesta.packages.geo_utils.geocode import generate_composite_key
+
+
+@contextmanager
+def crunchbase_tar():
+    """Downloads the tar archive of Crunchbase data.
+
+    Returns:
+        :code:`tarfile.Tarfile`: opened tar archive
+    """
+    crunchbase_config = misctools.get_config('crunchbase.config', 'crunchbase')
+    user_key = crunchbase_config['user_key']
+    url = 'https://api.crunchbase.com/v3.1/csv_export/csv_export.tar.gz?user_key='
+    with NamedTemporaryFile() as tmp_file:
+        r = requests.get(''.join([url, user_key]))
+        tmp_file.write(r.content)
+        try:
+            tmp_tar = tarfile.open(tmp_file.name)
+            yield tmp_tar
+        finally:
+            tmp_tar.close()
+
+
+def get_csv_list():
+    """Gets a list of csv files within the Crunchbase tar archive.
+
+    Returns:
+        list: all .csv files int the archive
+    """
+    csvs = []
+    csv_pattern = re.compile(r'^(.*)\.csv$')
+    with crunchbase_tar() as tar:
+        names = tar.getnames()
+    for name in names:
+        tablename = csv_pattern.match(name)
+        if tablename is not None:
+            csvs.append(tablename.group(1))
+    return csvs
+
+
+def get_files_from_tar(files, test=False):
+    """Converts csv files in the crunchbase tar into dataframes and returns them.
+
+    Args:
+        files (list): names of the files to extract (without .csv suffix)
+        test (bool): flag to reduce the number of records retrieved, for testing
+
+    Returns:
+        (:obj:`list` of :obj:`pandas.Dataframe`): the extracted files as dataframes
+    """
+    if type(files) != list:
+        raise TypeError("Files must be provided as a list")
+
+    nrows = 1000 if test else None
+    dfs = []
+    with crunchbase_tar() as tar:
+        for filename in files:
+            dfs.append(pd.read_csv(tar.extractfile(''.join([filename, '.csv'])),
+                                   low_memory=False, nrows=nrows))
+            logging.warning(f"Collected {filename} from crunchbase tarfile")
+    return dfs
+
+
+def rename_uuid_columns(data):
+    """Renames any columns called or containing `uuid`, to the convention of `id`.
+
+    Args:
+        data (:obj:`pandas.Dataframe`): dataframe with column names to amend
+
+    Returns:
+        (:obj:`pandas.Dataframe`): the original dataframe with amended column names
+    """
+    renames = {col: col.replace('uuid', 'id') for col in data}
+    return data.rename(columns=renames)
+
+
+def bool_convert(value):
+    """Converter from string representations of bool to python bools.
+
+    Args:
+        value (str): t or f
+
+    Returns:
+        (bool): boolean representation of the field, or None on failure
+    """
+    lookup = {'t': True, 'f': False}
+    try:
+        return lookup[value]
+    except KeyError:
+        logging.info(f"Not able to convert to a boolean: {value}")
+
+
+def _generate_composite_key(**kwargs):
```

Comment:
generate_composite_key is imported, but I felt this behaviour was specific to this package and not necessarily the way it would always want to be used